pkg/trie/cache/inmemory/lru_cache.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package inmemory
+
+import (
+	"github.com/karlseguin/ccache/v3"
```

Comment:
Gossamer currently have a LRU cache implementation at `lib/utils/lru-cache`, can it be used?

Reply:
Good catch, let me check, with that we can prevent adding new external deps! thks

Reply:
The difference is the way it calculates the capacity, I need a "custom" capacity using the value size instead of the amount of entries. So I can't just replace it to use the one we already have 😞

Reply:
I see, yeah, I don't mind to keep using ccache! Thanks for checking 👍